Fix libc heap size calculation for ESP32, use heap adapter

marekmatej opened the pull request:
This PR:
- fixes the libc heap allocation for ESP32 chips
- remove ESP heap and transform the heap adapter
- use kernel heap by default in RF stacks

thedjnK commented on lib/libc/common/Kconfig:
```diff
@@ -58,6 +58,7 @@ config COMMON_LIBC_MALLOC_ARENA_SIZE
 	int "Size of the common C library malloc arena"
 	depends on COMMON_LIBC_MALLOC
 	default 0 if MINIMAL_LIBC
+	default 2048 if PICOLIBC
```
nack, this works fine for every other system and -1 is what people expect, the esp memory should be the one that changes

marekmatej replied:
> nack, this works fine for every other system and -1 is what people expect, the esp memory should be the one that changes

I agree! I put this PR only to test the impact. I'll update soon. And thanks for taking a look @thedjnK

marekmatej committed:
lib: libc: malloc.c: fix calculation of sys heap

Change the condition so that ESP32 SoCs are using `_heap_sentry` to
get actual run-time heap size.

Signed-off-by: Marek Matej <marek.matej@espressif.com>

marekmatej committed:
soc: espressif: Remove ESP heap and use heap adapter

Remove ESP heap from the sources. System heap is default heap.
Use heap adapter layer to configure used heap.
Use MEM_POOL memory request config to Wi-Fi and Bluetooth drivers.
Update the Wi-Fi and BLE memory needs.

Signed-off-by: Marek Matej <marek.matej@espressif.com>

sylvioalves commented on drivers/wifi/esp32/Kconfig.esp32:
```diff
@@ -341,6 +341,9 @@ choice ESP_WIFI_HEAP
 	config ESP_WIFI_HEAP_SYSTEM
 		bool "Wifi adapter use system heap"
 
+	config ESP_WIFI_HEAP_KERNEL
```
Would you rename those `wifi ` text entries to `Wi-Fi` instead?

marekmatej committed:
manifest: update hal_espressif

Use proper hal version.

Signed-off-by: Marek Matej <marek.matej@espressif.com>

marekmatej commented:
@stephanosio can you PTAL

sylvioalves commented on drivers/wifi/esp32/Kconfig.esp32:
```diff
@@ -20,12 +20,24 @@ menuconfig WIFI_ESP32
 
+choice ESP_WIFI_HEAP
+	prompt "Wi-Fi adapter heap memory"
+
```
Please, set default option here:
`default ESP_WIFI_HEAP_SYSTEM`
